sandbox/apparmor/apparmor.go

```diff
@@ -316,27 +321,38 @@ func probeKernelFeatures() ([]string, error) {
 }
 
 func probeParserFeatures() ([]string, error) {
-	parser, err := findAppArmorParser()
+	parser, internal, err := FindAppArmorParser()
 	if err != nil {
 		return []string{}, err
 	}
 	features := make([]string, 0, 1)
 	if tryAppArmorParserFeature(parser, "change_profile unsafe /**,") {
 		features = append(features, "unsafe")
 	}
+	if internal {
+		features = append(features, "snapd-internal")
+	}
 	sort.Strings(features)
 	return features, nil
 }
 
-// findAppArmorParser returns the path of the apparmor_parser binary if one is found.
-func findAppArmorParser() (string, error) {
+// FindAppArmorParser returns the path of the apparmor_parser binary if one is found.
+func FindAppArmorParser() (string, bool, error) {
+	// first see if we have our own internal copy
+	if path, err := snapdtool.InternalToolPath("apparmor_parser"); err == nil {
```

Comment:
so if I understand correctly, if we are executing from `/usr/libexec/snapd/snapd` on i.e. fedora which disables re-exec, snapd is from the rpm package of snapd and this will only check for apparmor_parser existing in `/usr/libexec/snapd/apparmor_parser`, which will probably never exist (note this assumes that apparmor is a thing on fedora which AIUI, it isn't but just for the sake of example). And specifically this InternalToolPath will not search the snapd snap since snapd is not running from the snapd snap.

I think that's the expected and desired behavior, as on these other distros which specifically have disabled re-exec, we want to use the apparmor_parser that is on the $PATH and not from inside the snapd snap.

Reply:
Yes this is designed to prefer the snapd-snap's internal version of apparmor_parser if it is present, but otherwise fall-back to using the one from PATH so that we don't break this existing use-case.

Reply:
great, thanks for confirming